### PR TITLE
Infinite integer

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
Implement Id class
    
Closes: https://github.com/metarhia/common/issues/161

### CODE REVIEW

**Comment by SemenchenkoVitaliy** [comment]
Btw, is there a better way of testing class methods? And is it ok to take repeated data outside of test functions?

File: lib/id.js [hunk 1]
```diff
@@ -139,14 +139,96 @@ const idToPath = id => {
...
   }
   return parseInt(hex, 16);
 };
 
+class Id {
+  constructor(
+    n // Initial value
+    // string or number or Array of Number, Uint32Array
+  ) {
+    if (typeof(n) === 'number')
```
Comment (tshemsedinov) [inline lib/id.js]: If we use `else if` we need block statements

**Comment by SemenchenkoVitaliy** [comment]
Also, we need to update .eslintrc.yml to be able to use BigInt and to break tests

File: lib/id.js [hunk 2]
```diff
@@ -139,14 +139,96 @@ const idToPath = id => {
...
+      this.value = n;
+    else if (typeof(n) === 'string')
+      this.value = parseInt(n, 10) || 0;
+    else if (Array.isArray(n) &&
```
Comment (tshemsedinov) [inline lib/id.js]: It will be better to put conditions in single line

File: lib/id.js [hunk 3]
```diff
@@ -139,14 +139,96 @@ const idToPath = id => {
...
 
+class Id {
+  constructor(
+    n // Initial value
+    // string or number or Array of Number, Uint32Array
+  ) {
+    if (typeof(n) === 'number')
+      this.value = n;
+    else if (typeof(n) === 'string')
+      this.value = parseInt(n, 10) || 0;
```
Comment (tshemsedinov) [inline lib/id.js]: Not `0` better `NaN` or `undefined`

File: lib/id.js [hunk 4]
```diff
@@ -139,14 +139,96 @@ const idToPath = id => {
...
+  }
+
+  toString(
+    // Get current id value
+    // Returns: string
+  ) {
+    return this.value + '';
+  }
+
+  next(
```
Comment (tshemsedinov) [inline lib/id.js]: Need no `next`

File: lib/id.js [hunk 5]
```diff
@@ -139,14 +139,96 @@ const idToPath = id => {
...
+    // Increment id value by 1
+    // Returns: number
+  ) {
+    return ++this.value;
+  }
+
+  set(
```
Comment (tshemsedinov) [inline lib/id.js]: Need no `set`

**Comment by tshemsedinov** [comment]
Closed without landing, here is explanation: https://github.com/metarhia/common/issues/161

**Pull request closed without merging**